### PR TITLE
Add back a method called by TypeScript to track async operations.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/Features/Core/Portable/Shared/TestHooks/IAsynchronousOperationListener.cs [hunk 1]
```diff
@@ -6,6 +6,10 @@ namespace Microsoft.CodeAnalysis.Shared.TestHooks
 {
     internal interface IAsynchronousOperationListener
     {
+        // TODO(cyrusn): Remove this overload when we can simply recompile TypeScript and move them
```
Comment (jaredpar) [inline src/Features/Core/Portable/Shared/TestHooks/IAsynchronousOperationListener.cs]: Can we file a bug for this?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Shared/TestHooks/IAsynchronousOperationListener.cs]: Sure.

**Comment by CyrusNajmabadi** [comment]
Bug here: https://github.com/dotnet/roslyn/issues/6640

**Comment by CyrusNajmabadi** [comment]
@jaredpar Are you ok with this change going in?

**Comment by jaredpar** [comment]
@CyrusNajmabadi yes so long as there is a follow up change that converts TODO -> bug.

**Comment by Pilchie** [comment]
The original change didn't make it into stabilization, right?  There isn't an issue with TypeScript there?

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**